### DISCUSSION TITLE
Laravel Mix assets url conflicts

### DISCUSSION THREAD

**Discussion by clementmas** [laravel/nova-issues#4104] [Ideas & Feature Requests]
### Description:

Laravel Nova (v4.2.5) is loading its assets using the `mix()` helper (see `nova\resources\views\layout.blade.php` below), which creates conflicts when [`mix_url`](https://laravel.com/docs/9.x/mix#custom-mix-base-urls) is set to a different domain name.

```
<!-- Styles -->
<link rel="stylesheet" href="{{ mix('app.css', 'vendor/nova') }}">

<!-- Scripts -->
<script src="{{ mix('manifest.js', 'vendor/nova') }}"></script>
<script src="{{ mix('vendor.js', 'vendor/nova') }}"></script>
<script src="{{ mix('app.js', 'vendor/nova') }}"></script>
```

### Detailed steps to reproduce the issue on a fresh Nova installation:

- Update `config/app.php`, add `'mix_url' => 'https://cdn.laravel.test'`

=> Nova's assets (JS, CSS and fonts) can't be loaded.

This means that at the moment, Laravel Nova can't be added to an existing Laravel application that has its own custom Laravel Mix config.

To prevent this conflit, Nova's assets should be loaded from an absolute url `/vendor/nova/...` or using the value from `config('nova.domain')`.

**Comment by clementmas** [comment]
@crynobone this is not a "feature request". It is a bug. Please reopen the issue.

I bought Laravel Nova to add it to an existing Laravel app. The documention doesn't mention any incompatibilities, because there shouldn't be. Nova's assets have nothing to do with the default assets.

**Reply by crynobone** [reply]
If you want to use CDN, you should upload all public assets to CDN. 

We not going to simply change and make Laravel Nova assets handling worst simply just from this request. Also, `mix()` isn't something we added in Nova 4. It has been around since the first commit.

**Reply by clementmas** [reply]
Thanks for clarifying your position. I don't think issues should be closed without any explanation.

I still think this is an issue and not a "request". The pipeline to compile Laravel's assets and Nova's assets are different so it makes sense to use or propose a different URL generation logic. I don't see how this is worse.

Basically if `mix_url` is set to any other domain than Nova's domain, there will be issues. Including CORS issues when fonts are loaded from a different subdomain. And I think that should at least be documented, if not supported.

But I won't insist. You don't seem open to have a conversation about it.
